tastypie/resources.py

```diff
@@ -2091,8 +2091,6 @@ def lookup_kwargs_with_identifiers(self, bundle, kwargs):
         """
         lookup_kwargs = {}
         bundle.obj = self.get_object_list(bundle.request).model()
-        # Override data values, we rely on uri identifiers
-        bundle.data.update(kwargs)
```

Comment:
We can't remove this line. If 'pk' is showing up in your data we'll need to remove it elsewhere.